packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -216,11 +235,26 @@ function patchedObserve(
     if (isUndefined(target[observerLookupField])) {
         defineProperty(target, observerLookupField, { value: [] });
     }
-    ArrayPush.call(target[observerLookupField], this);
+    // Same observer trying to observe the same node
+    if (ArrayIndexOf.call(target[observerLookupField], this) === -1) {
+        ArrayPush.call(target[observerLookupField], this);
+    } // else There is more bookkeeping to do here https://dom.spec.whatwg.org/#dom-mutationobserver-observe Step #7
+
     // If the target is a SyntheticShadowRoot, observe the host since the shadowRoot is an empty documentFragment
     if (target instanceof SyntheticShadowRoot) {
         target = (target as ShadowRoot).host;
     }
+
+    // maintain a list of all nodes observed by this observer
+    if (observerToNodesMap.has(this)) {
+        const observedNodes = observerToNodesMap.get(this)!;
+        if (observedNodes.indexOf(target) === -1) {
+            observedNodes.push(target);
+        }
+    } else {
+        observerToNodesMap.set(this, [target]);
```

Comment:
set from lang, although we are not doing that today... we should probably start doing it at some point.

Reply:
Its a polyfilled feature in IE11, wasn't sure if was okay to cache these. Let me know.

Reply:
yes, it should be ok. they all run before everything else.

Reply:
lets open a seprate issue for this in general.

Reply:
https://github.com/salesforce/lwc/issues/1427